docs/rules/valid-typeof.md

```diff
@@ -28,6 +34,24 @@ typeof foo === baz
 typeof bar === typeof qux
 ```
 
+Examples of **incorrect** code with the `{ "requireStringLiterals": true }` option:
+
+```js
+typeof foo === undefined
+typeof bar == Object
+typeof baz === "strnig"
+typeof foo === 5
+typeof bar === typeof qux
+```
+
+Examples of **correct** code with the `{ "requireStringLiterals": true }` option:
+
+```js
+typeof foo === "undefined"
+typeof bar == "object"
+typeof baz === "string"
```

Comment:
Wondering if we want to mention something like `typeof thing = "foo"`? To clarify that we're not checking if it's a valid string literal.